cheroot/server.py

```diff
@@ -1464,20 +1480,6 @@ def peer_group(self):
         _, group = self.resolve_peer_creds()
         return group
 
-    def _close_kernel_socket(self):
-        """Close kernel socket in outdated Python versions.
-
-        On old Python versions,
-        Python's socket module does NOT call close on the kernel
-        socket when you call socket.close(). We do so manually here
-        because we want this server to send a FIN TCP segment
-        immediately. Note this must be called *before* calling
-        socket.close(), because the latter drops its reference to
-        the kernel socket.
```

Comment:
Yes, this is the original commit where this change was introduced https://github.com/cyraxjoe/cheroot/commit/b8d0865cabad55eb6bb52cd84d695d1731acdfdf, I remember stepping on the debugger while having a wireshark session open to make sure the connection was closed at the _OS_ level with shutdown. As for the special case for pyopenssl... I based that decision solely on pyopenssl docs, the plain `shutdown` exposed doesn't receive any arguments, later I found that the approach of using that method instead of the one for the underlying socket was also used in https://github.com/cherrypy/cheroot/blob/master/cheroot/workers/threadpool.py#L300-L314. Alternatively we could also use that here, and verify that it works with tcpdump or wireshark.

Reply:
The main motivation of this question is to ensure there's no regression since the change is not annotated with the explanation comments confirming this.